lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -812,14 +812,23 @@ static bool isExpressionUnableToInline(Operation *op,
           !isOkToBitSelectFrom(op->getResult(0)))
         return true;
 
-    // Always blocks must have a name in their sensitivity list, not an expr.
-    if (!options.allowExprInEventControl && isa<AlwaysOp, AlwaysFFOp>(user)) {
-      // Anything other than a read of a wire must be out of line.
-      if (auto read = dyn_cast<ReadInOutOp>(op))
-        if (read.getInput().getDefiningOp<sv::WireOp>() ||
-            read.getInput().getDefiningOp<RegOp>())
-          continue;
-      return true;
+    // Handle option disallowing expressions in event control.
+    if (!options.allowExprInEventControl) {
+      // "disable iff" condition shouldn't be inlined.
+      if (auto disableOp = dyn_cast<ltl::DisableOp>(user);
+          disableOp && disableOp.getCondition().getDefiningOp() == op)
+        return true;
```

Comment:
Ah, looking at the code example below, I think this can crash.

Reply:
Thanks for helping vet this carefully! I'm not sure I follow about why this might crash -- is the concern that getDefiningOp() would return null?

I'll push a rework in a few to pickup same thing for ClockOp too and to avoid rejecting "inline" here if it is a read of a wire (like done for AlwaysOp presently).

Reply:
Why must the use be the condition? Since we can't spill the other operands to wires (AFAIK), at least not when they're "Property" values themselves, this check is there both for precision but also to avoid attempting to request spilling an LTL property itself to a wire.

It wasn't obvious to me this code is only invoked on operations that only produce results that are HW values, but maybe that's a pre-condition?

A good question is whether being used for the "property" operand of a disable is okay re:this linting rule, not sure.  Will push in a second